homeassistant/components/binary_sensor/xiaomi_aqara.py

```diff
@@ -311,6 +316,48 @@ def parse_data(self, data, raw_data):
             return False
 
 
+class XiaomiVibration(XiaomiBinarySensor):
+    """Representation of a Xiaomi Vibration Sensor."""
+
+    def __init__(self, device, name, data_key, hass, xiaomi_hub):
+        """Initialize the XiaomiVibration."""
+        self._hass = hass
+        self._last_action = None
+        XiaomiBinarySensor.__init__(self, device, name, xiaomi_hub,
+                                    data_key, None)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {ATTR_LAST_ACTION: self._last_action}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
+    def parse_data(self, data, raw_data):
+        """Parse data sent by gateway."""
+        value = data.get(self._data_key)
+        if value is None:
+            return False
+
+        if value == 'vibrate':
+            movement_type = 'vibrate'
+        elif value == 'tilt':
+            movement_type = 'tilt'
+        elif value == "free_fall":
+            movement_type = 'free_fall'
+        else:
+            _LOGGER.warning("Unsupported movement_type detected: %s", value)
+            return False
+
+        self._hass.bus.fire('movement', {
```

Comment:
Use `self.hass`. Use a platform specific event type name.

Reply:
This has been done. Please note, this class now varies greatly from the rest of the classes for this platform. I'm firing xiaomi_aqara.movement, others are firing "click", "cube", and "motion". Changing the others would represent a breaking change. 

Additionally, this is now the only class in this platform that is using the "hass" that is set on the Entity automatically. All other classes are still using a passed in "hass".